tests/instructions/mul.rs

```diff
@@ -471,4 +471,124 @@ mod mul_tests {
+            vmtropy::execute_function(&program, function_name, &user_inputs).unwrap();
+
+        let expected_function_variables = vec!["r0", "r1", "r2"];
+        for (register, expected_register) in
+            function_variables.keys().zip(expected_function_variables)
+        {
+            assert_eq!(register, expected_register);
+        }
+
+        let r0 = function_variables["r0"].as_ref().unwrap();
+        assert!(matches!(r0, vmtropy::CircuitIOType::SimpleInt8(_)));
+        assert_eq!(r0.value().unwrap(), "1".to_owned());
+
+        let r1 = function_variables["r1"].as_ref().unwrap();
+        assert!(matches!(r1, vmtropy::CircuitIOType::SimpleInt8(_)));
+        assert_eq!(r1.value().unwrap(), "1".to_owned());
+
+        let r2 = function_variables["r2"].as_ref().unwrap();
+        assert!(matches!(r2, vmtropy::CircuitIOType::SimpleInt8(_)));
+        assert_eq!(r2.value().unwrap(), "1".to_owned());
+    }
+
+    #[test]
+    fn test_mul_with_i8_private_inputs() {
+        let program_string = test_helpers::read_program("mul").unwrap();
+        let (_, program) = Program::<Testnet3>::parse(&program_string).unwrap();
+        let function_name = "hello_14";
+
+        /*
+        function hello_8:
+            input r0 as i8.private;
+            input r1 as i8.private;
+            mul r0 r1 into r2;
+            output r2 as i8.private;
+        */
+
+        let user_inputs = vec![I8(1), I8(1)];
+
+        // execute circuit
+        let (function_variables, _proof) =
+            vmtropy::execute_function(&program, function_name, &user_inputs).unwrap();
+
+        let expected_function_variables = vec!["r0", "r1", "r2"];
+        for (register, expected_register) in
+            function_variables.keys().zip(expected_function_variables)
+        {
+            assert_eq!(register, expected_register);
+        }
+
+        let r0 = function_variables["r0"].as_ref().unwrap();
+        assert!(matches!(r0, vmtropy::CircuitIOType::SimpleInt8(_)));
+        assert_eq!(r0.value().unwrap(), "1".to_owned());
+
+        let r1 = function_variables["r1"].as_ref().unwrap();
+        assert!(matches!(r1, vmtropy::CircuitIOType::SimpleInt8(_)));
+        assert_eq!(r1.value().unwrap(), "1".to_owned());
+
+        let r2 = function_variables["r2"].as_ref().unwrap();
+        assert!(matches!(r2, vmtropy::CircuitIOType::SimpleInt8(_)));
+        assert_eq!(r2.value().unwrap(), "1".to_owned());
+    }
+
+    #[test]
+    fn test_mul_with_i8_private_and_public_inputs() {
+        let program_string = test_helpers::read_program("mul").unwrap();
+        let (_, program) = Program::<Testnet3>::parse(&program_string).unwrap();
+        let function_name = "hello_15";
+
+        /*
+        function hello_15:
+            input r0 as i8.public;
+            input r1 as i8.public;
+            mul r0 r1 into r2;
+            output r2 as i8.private;
+        */
+
+        let user_inputs = vec![I8(1), I8(1)];
```

Comment:
Maybe we should make the arguments of these tests differ a little bit more (like in the instructions tests), in order to avoid what had happened with `div`. Not necessary at all for this PR but something to keep in mind.